src/main/java/co/omise/Client.java

```diff
@@ -67,19 +75,22 @@ public Client(String secretKey) throws ClientException {
      * Creates a Client that sends the specified API version string in the header to access an earlier version
      * of the Omise API.
      *
-     * @param publicKey  The key with {@code pkey_} prefix.
-     * @param secretKey  The key with {@code skey_} prefix.
+     * @param publicKey The key with {@code pkey_} prefix.
+     * @param secretKey The key with {@code skey_} prefix.
      * @throws ClientException if client configuration fails (e.g. when TLSv1.2 is not supported)
+     * @see Serializer
      * @see <a href="https://www.omise.co/security-best-practices">Security Best Practices</a>
      * @see <a href="https://www.omise.co/api-versioning">Versioning</a>
      */
     public Client(String publicKey, String secretKey) throws ClientException {
         Preconditions.checkNotNull(secretKey);
 
-        config = new Config(Endpoint.API_VERSION, publicKey, secretKey);
+        Config config = new Config(Endpoint.API_VERSION, publicKey, secretKey);
         httpClient = buildHttpClient(config);
 
-        account = new AccountResource(httpClient);
+        Serializer serializer = Serializer.defaultSerializer();
+        requester = new RequesterImpl(httpClient, serializer);
```

Comment:
@Namiii possible to allow configuration of the implementation to use so that we can add app engine support in the future?

Reply:
```java
public Client(String publicKey, String secretKey, Requester requester) {
```

Reply:
@chakrit that would mean the user would need to create their own Requester, which in turn means creating their own HttpClient and Serializer, and also they'll need to have their own concrete implementation of Requester interface, would that work well? I'm not really sure

Reply:
@chakrit that would mean the user would need to create their own Requester, which in turn means creating their own HttpClient and Serializer, I'm not really sure about that

Reply:
@Namiii in the long term we could maybe provide `AbstractRequester` so that it's more idiomatic java. but for now just wanna have answer for when people come asking for App Engine support.

Reply:
Alright I get that it'll be an overloaded constructor, but you think it'll be ok for them to handle all the initializations and implementation for the `Requester`?  Just trying to think if that'll cause issues

Reply:
@Namiii I think it's OK for now. we can make it easier as we progress but at least it's possible now.

Reply:
So impossible -> possible -> easy. Now we at least move from impossible to possible. We can make it easy in v3.1 or later.

Reply:
@chakrit You're right, need to look at the bigger picture! 👌Will continue with the rest using this same format